### PR TITLE
Activities for calendars, events and todos

### PR DESCRIPTION
**Pull request by nickvergessen**
Supports:
- add/update/delete/share/unshare of calendars
- add/update/delete of events
- add/update/delete/mark as solved/reopen of todos

Todos:
- [x] Dispatcher
- [x] Tests

Ref #745 

@MorrisJobke @LukasReschke @rullzer please review :)

### CODE REVIEW

File: apps/dav/tests/unit/CalDAV/CalDavBackendTest.php [hunk 1]
```diff
@@ -357,10 +357,8 @@ public function testPublications() {
 		$calendar->setPublishStatus(false);
 		$this->assertEquals(false, $calendar->getPublishStatus());
 
-		$publicCalendarURI = md5($this->config->getSystemValue('secret', '') . $calendar->getResourceId());
```
Comment (tcitworld) [inline apps/dav/tests/unit/CalDAV/CalDavBackendTest.php]: :+1:

**Comment by MorrisJobke** [comment]
> Tests

cc @nickvergessen

**Comment by nickvergessen** [comment]
Added a bunch of tests, will add the missing tests later in another PR.
Can we merge this?

**Comment by MorrisJobke** [comment]
Tested and works - but only with https://github.com/nextcloud/activity/pull/64

:+1:

**Comment by MorrisJobke** [comment]
cc @rullzer - review time

**Comment by rullzer** [comment]
LGTM

**Comment by nickvergessen** [comment]
@mmluqman when you switch to Nextcloud, you can have activities for calendar and task changes already since version 11. I just saw you commented on the ownCloud feature request for it. So I thought you might be interested in it ;)

**Comment by jospoortvliet** [comment]
same for @niko886 -> this allows you to get notified on calendar changes (new/modified). Not yet for when an event happens though, see discussion about that here: https://github.com/nextcloud/server/issues/1642

(one question is: how do you avoid notifications by multiple apps. I guess it just needs to be configurable...)

**Pull request merged**